libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -178,25 +177,21 @@ ReturnCode SoftwareContainer::startGateways(const GatewayConfiguration &gwConfig
         throw InvalidOperationError(message);
     }
 
-    ReturnCode result = ReturnCode::SUCCESS;
-
-    result = configureGateways(gwConfig);
-    if (isError(result)) {
-        return result;
+    if (!configureGateways(gwConfig)) {
+        return false;
     }
 
-    result = activateGateways();
-    if (isError(result)) {
-        return result;
+    if (!activateGateways()) {
+        return false;
```

Comment:
You could combine configureGateways() and activateGateways() as follows, since or is lazy.

```
if (!configureGateways() || !activateGateways()) {
    return false;
}
```